go/vt/sqlparser/ast_rewriting.go

```diff
@@ -255,12 +263,20 @@ type expressionRewriter struct {
 	// we need to know this to make a decision if we can safely rewrite JOIN USING => JOIN ON
 	hasStarInSelect bool
 
-	keyspace    string
-	selectLimit int
+	keyspace      string
+	selectLimit   int
+	setVarComment string
+	sysVars       map[string]string
 }
 
-func newExpressionRewriter(keyspace string, selectLimit int) *expressionRewriter {
-	return &expressionRewriter{bindVars: &BindVarNeeds{}, keyspace: keyspace, selectLimit: selectLimit}
+func newExpressionRewriter(keyspace string, selectLimit int, setVarComment string, sysVars map[string]string) *expressionRewriter {
```

Comment:
nit: I know this isn't the first thing that `expressionRewriter` is responsible for that is not an expression. Still - maybe now the time has finally come to rename this struct to something broader than expression?

Reply:
that's a good point, I renamed it to `astRewriter` since we're basically rewriting most of the AST